src/main/java/frc/team281/robot/ConvertRectToPoint.java

```diff
@@ -1,121 +1,128 @@
+    public static  Point[] concatenate(Point[] a, Point[] b) 
+    { 
+        Point[] combine = new Point[8];
+
+        for (int i=0; i<8; i++){
+            if (i<4){
+                combine[i] = a[i];
+            }
+            else{
+                combine[i] = b[i-4];
+            }
+        }
         
-        double h=first.size.width;
-        double w=first.size.height;
-        double hyp = Math.sqrt(h*h + w*w);
-        double c= Math.PI/180;
-        double total_angle_radian=first.angle+Math.asin(w/h);
-        double angle_radian = first.angle;
-
-        //angle of rotated rect is in refrence to vertical
-
-        double first_x=first.center.x-Math.sin(total_angle_radian)*hyp/2;
-        double first_y=first.center.y-Math.cos(total_angle_radian)*hyp/2;
-
-        //Starts at rightmost point and goes clockwise ending at uppermost point
-
-        arraypoints.add(new Point(first_x, first_y));
-        arraypoints.add(
-            new Point(
-                first_x+w*Math.sin(angle_radian), 
-                first_y+w*Math.cos(angle_radian)
-                )
-            );
-        arraypoints.add(
-            new Point(
-                first_x+hyp*Math.sin(total_angle_radian), 
-                first_y+hyp*Math.cos(total_angle_radian)
-                )
-            );
-        arraypoints.add(
-            new Point(
-                first_x+h*Math.sin(angle_radian), 
-                first_y-h*Math.cos(angle_radian)
-                )
-            );
-        return arraypoints;
-    } 
-
-    public boolean isFirstLeftMost( RotatedRect first, RotatedRect second){
+
+        return combine;
+
+    }
+
+public static boolean isFirstLeftMost( RotatedRect first, RotatedRect second){
         if (first.center.x<second.center.x){
             return true;
         }else{
             return false;
         }
 
     }
-}
\ No newline at end of file
+}
+
+
+    /*public static ArrayList<Point> centerAtOrigin(RotatedRect rect) {
+        //Order: Top right, bottom right, bottom left, top left
+        double height = rect.size.height;
+        double width = rect.size.width;
+
+        Point topRight = new Point((0.5)*height,(0.5)*width);
+        Point bottomRight = new Point((0.5)* height,-((0.5)*width));
+        Point bottomLeft = new Point(-((0.5)* height),-(0.5)*width);
+        Point topLeft = new Point(-(0.5)* height,(0.5)*width);
+        
+        var vertices = new ArrayList<Point>();
+        vertices.add(topRight);
+        vertices.add(bottomRight);
+        vertices.add(bottomLeft);
+        vertices.add(topLeft);
+
+        return vertices;
+    }
+
+    public static ArrayList<Point> rotateAroundOrigin(ArrayList<Point> vertices, RotatedRect rect) {
+        
+        for(Point p: vertices){
+            DecimalFormat df = new DecimalFormat("#, ###.##");
+            String xPrime = df.format(p.x * Math.cos(Math.PI-rect.angle) - p.y * Math.sin(Math.PI-rect.angle));
```

Comment:
I think you have problems here.  RotatedRect.angle is measured in degrees while Math.cos etc want angles in radians.  You need to take your rect.angle and multiple by (Math.PI/180.0)

Reply:
From: https://docs.opencv.org/3.4/db/dd6/classcv_1_1RotatedRect.html
"Each rectangle is specified by the center point (mass center), length of each side (represented by Size2f structure) and the rotation angle in degrees."